src/ch00-08-function_attributes.md

```diff
@@ -0,0 +1,17 @@
+# Function Attributes
+
+When you write functions in a contract using Cairo, you need to provide special annotations to inform the compiler about the nature of each function. These annotations are written as `#[attribute]` just before the function definition.
+
+Let's explore the three main types of functions you can use:
+
+- `#[external]`: These functions can be called from anywhere and have the ability to modify the contract's state. However, keep in mind that calling them incurs gas costs.
```

Comment:
Small misconception here:
> These functions can be called from anywhere and have the ability to modify the contract's state. However, keep in mind that calling them incurs gas costs.

You can only interact with `external` functions by sending a transaction. That's why there are associated gas costs, because the sequencer has to execute an operation. However:

> These functions can also be called from anywhere, but they are read-only and cannot modify the contract's state. The best part is that they are free to call!

View functions are free to call when you interact with them through an RPC node directly, without sending a transaction. But if you're sending a transaction, in which a call to a `view` function of a contract is made, the execution of this function is not going to be free!

Reply:
I see. I may have taken a shortcut here and I'll include these details. Thank you!